packages/web/src/Components/TextField.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { FunctionComponent } from 'react';
+import styled from 'styled-components';
+
+import { withTheme } from '@material-ui/styles';
+import { IThemeWeb } from 'Themes';
+
+interface ITextFieldProps {
+  [propName: string]: any;
+  /**
+   * The input id property
+   */
+  id: string;
+  /**
+   * @optional
+   */
+  theme?: IThemeWeb | unknown;
+  /**
+   * @optional
+   * Help text placed underneath the text TextField
+   */
+  label?: string;
+  /**
+   * @optional
+   * Help text placed underneath the text TextField
+   */
+  helpText?: string;
+  /**
+   * @optional
+   * If this field is required or not
+   */
+  required?: boolean;
+  /**
+   * @optional
+   * Disables the TextField interaction
+   */
+  disabled?: boolean;
+  /**
+   * @optional
+   * Optional status for component variant
+   */
+  status?: 'error' | 'success' | undefined;
+  /**
+   * @optional
+   * Optional input type property
+   */
+  type?: 'text' | 'password' | 'search' | 'email' | string;
```

Comment:
Type could be used as an interface extension like `extends Pick<InputBaseProps, 'type'>`